### PR TITLE
fix: resolve symlinked files

### LINKED ISSUE #190: Bug: Resources README doesn't always render correctly
See [Remix Fastify](https://remix.run/resources/remix-fastify) for an example

When I wrote the README fetcher for the Resource pages, I took a [naive approach](https://github.com/remix-run/remix-website/blob/main/app/lib/resources.server.ts#L107-L114)

Basically I'm assuming that the README url is always `main/readme.md`. This is not always true (see the Remix Fastify example).

The good thing is `octokit` will give you the appropriate URL with the following

```js
context.octokit.rest.repos.getReadme({ owner, repo })
```

All that's required is a little refactoring of code to fetch the correct content and the rendering should take care of itself

### PR DESCRIPTION
**Pull request by mcansh**
mcansh/remix-fastify root README is a symlink to the README in packages/remix-fastify and using the raw.github... doesnt follow the redirect

closes #190

feel free to close if any of the other commenters on the issue also do the work so they can get the contribution :)

Signed-off-by: Logan McAnsh <logan@mcan.sh>

### CODE REVIEW

File: app/lib/gh-docs/repo-content.ts [hunk 1]
```diff
@@ -17,13 +18,25 @@ export async function getRepoContent(
...
   if (ref === "local") return getLocalContent(filepath);
   let [owner, repo] = repoPair.split("/");
-  let pathname = `/${owner}/${repo}/${ref}/${filepath}`;
-  let response = await fetch(
-    new URL(pathname, "https://raw.githubusercontent.com/").href,
-    { headers: { "User-Agent": `docs:${owner}/${repo}` } },
-  );
-  if (!response.ok) return null;
-  return response.text();
+  let contents = await octokit.rest.repos.getContent({
```
Comment (brookslybrand) [inline app/lib/gh-docs/repo-content.ts:21]: This is probably fine to pull in `octokit` directly. However, I noticed that in [some other files](https://github.com/remix-run/remix-website/blob/main/app/lib/gh-docs/tags.ts#L16_ we pass `octokit` in as an argument. Do you know if there was any specific reason for this pattern initially?

Comment (mcansh) [reply app/lib/gh-docs/repo-content.ts:21]: i don't know if there was any reason for passing it, looks like chance made that decision sometime last year before he left. i can match the pattern if preferred.

File: app/lib/gh-docs/docs.ts [hunk 2]
```diff
@@ -81,19 +82,22 @@ function parseAttrs(
...
   ttl: NO_CACHE ? 1 : 1000 * 60 * 5, // 5 minutes
   allowStale: !NO_CACHE,
   noDeleteOnFetchRejection: true,
   fetchMethod: fetchDoc,
 });
 
-async function fetchDoc(key: string): Promise<Doc> {
+async function fetchDoc(
+  key: string,
+  _staleValue: Doc | undefined,
```
Comment (brookslybrand) [inline app/lib/gh-docs/docs.ts:95]: I'm not sure I understand why `_staleValue` is in here, doesn't seem like we use it. Also, when we call `docCache.fetch` we're not passing it in, am I missing something?

File: app/lib/resources.server.ts [hunk 3]
```diff
@@ -96,26 +95,35 @@ declare global {
...
+async function fetchReadme(
+  key: string,
+  _staleValue: string | undefined,
+  { context }: LRUCache.FetchOptionsWithContext<string, string, CacheContext>,
+): Promise<string> {
+  let [owner, repo] = key.split("/");
+  let contents = await context.octokit.rest.repos.getReadme({
+    owner,
+    repo,
+    mediaType: { format: "base64" },
```
Comment (brookslybrand) [inline app/lib/resources.server.ts:115]: I'm just now realizing that we can just make it `format: "raw"` 🤦. I feel like that makes sense, what do you think? Pretty sure raw will also be smaller in size and less work than getting a base64 encoding

File: app/lib/gh-docs/docs.ts (hunk 2 above)
Comment (mcansh) [reply app/lib/gh-docs/docs.ts:95]: it’s a required argument when using the LRU.FetchOptionsWithContext as we’re passing around an octokit instance, but yes, we’re not using it, it’s added by LRUCache. followed convention from other cache.fetch functions

Comment (brookslybrand) [reply app/lib/gh-docs/docs.ts:95]: Great, figured I was missing something, thanks for the explanation!

**Review by brookslybrand — APPROVED** [review]
Great improvements on top of a fix. Thanks @mcansh!

**Pull request merged**